# Inferring dynamic GRNs using kernel reweighted Gaussiangraphical model from single cell gene expression data
# Term Paper for CSE 8903 - Special Problem, Fall 2021 (Jongseok Han)

## Description
Inferring cell specific GRN using single-cell gene expression data generated by softODE simulator

* `src` stores the inference algorithms including baseline.
* `test` stores the running script `softODE_test` for Dynamic GRN inference
* `simulator` stores softODE simulator `soft_boolODE` and running script `run_simulator`
* `data` stores the synthetic expression data (data is not uploaded because it is too large, but available after running simulator in local)

## Dependency
(required)
pytorch >= 1.15.0 
numpy >= 1.19.5
scipy >= 1.7.1
networkx >= 2.5
sklearn >= 0.24.2

(optional)
matplotlib >= 3.4.3
statsmodels >= 0.12.2
```

## Usage
* Set the hyper-parameter: bandwidth $$/rho$$, neighborhoodsize, and lambda.
* Set the hyper-parameter including: bandwidth, neighborhoodsize, and lambda. e.g.
  ```python
  # smaller bandwidth means that GRN of cells are more heterogeneous.
  bandwidth = 1
  # number of neighbor being considered when calculating the covariance matrix.
  n_neigh = 30
  # sparsity regulatorization, larger lamb means sparser result.
  lamb = 0.1
  ```
